lib/battle/obstacle/SpellCreatedObstacle.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * SpellCreatedObstacle.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#include "StdInc.h"
+#include "SpellCreatedObstacle.h"
+#include "CTownHandler.h"
+#include "VCMI_Lib.h"
+#include "spells/CSpellHandler.h"
+#include "filesystem/ResourceID.h"
+#include "battle/obstacle/ObstacleJson.h"
+
+SpellCreatedObstacle::SpellCreatedObstacle()
+{
+	casterSide = -1;
+	spellLevel = -1;
+	casterSpellPower = -1;
+	turnsRemaining = -1;
+	visibleForAnotherSide = -1;
+}
+
+SpellCreatedObstacle::SpellCreatedObstacle(ObstacleJson info)
+{
+	setArea(ObstacleArea(info.getArea(), info.getPosition()));
+	spellLevel = info.getSpellLevel();
+	turnsRemaining = info.getTurnsRemaining();
+	casterSpellPower = info.getSpellPower();
+	visibleForAnotherSide = info.isVisibleForAnotherSide();
+	casterSide = info.getBattleSide();
+	setGraphicsInfo(info.getGraphicsInfo());
+}
+
+bool SpellCreatedObstacle::canRemovedBySpell() const
+{
+	switch (getType())
+	{
+	case ObstacleType::FIRE_WALL:
+		if(spellLevel >= 2)
```

Comment:
This is completely wrong. Result depends on remove obstacle spell level but not the level obstacle itself was created with.
It you want this logic be here, pass spellLevel from RemoveObstacleMechanics